apis/archiver/v1alpha1/types.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright AppsCode Inc. and Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	batch "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "kmodules.xyz/client-go/api/v1"
+	ofst "kmodules.xyz/offshoot-api/api/v1"
+	stashcoreapi "kubestash.dev/apimachinery/apis/core/v1alpha1"
+)
+
+// +kubebuilder:validation:Enum=CSISnapshotter
+type BackupDriver string
+
+const (
+	// Takes full backup using CSISnapshotter driver
+	BackupDriverCSISnapshotter BackupDriver = "CSISnapshotter"
+)
+
+type FullBackupOptions struct {
+	Driver BackupDriver `json:"driver"`
+	// +optional
+	CSISnapshotter *CSISnapshotterOptions `json:"csiSnapshotter,omitempty"`
+	// +optional
+	Scheduler *SchedulerOptions `json:"scheduler,omitempty"`
+	// +optional
+	ContainerRuntimeSettings *ofst.ContainerRuntimeSettings `json:"containerRuntimeSettings,omitempty"`
+	// +optional
+	JobTemplate *ofst.PodTemplateSpec `json:"jobTemplate,omitempty"`
+	// +optional
+	RetryConfig *stashcoreapi.RetryConfig `json:"retryConfig,omitempty"`
+	// +optional
+	Timeout *metav1.Duration `json:"timeout,omitempty"`
+	// +optional
+	SessionHistoryLimit int32 `json:"sessionHistoryLimit,omitempty"`
+}
+
+type ManifestBackupOptions struct {
+	// +optional
+	EncryptionSecret *v1.ObjectReference `json:"encryptionSecret"`
```

Comment:
Is it possible to encrypt for the fullDB backup ?  If possbie, we should add this field there .

Reply:
no